File: signature/perl.go
package signature

import (
	"regexp"
	"strings"

	"github.com/0n1shi/whopper/analyzer"
	"github.com/0n1shi/whopper/crawler"
)

var PerlSignature = Signature{
	Name:        "Perl",
	Description: "Perl is a highly capable, feature-rich programming language with over 37 years of development.",
	Cpe:         "cpe:/a:perl:perl",

	DetectPattern: DetectPattern{
		Headers: []Header{{
			Name:  "server",
			Value: "Perl",
		}},
	},
	VersionPattern: VersionPattern{
		Headers: []Header{{
			Name:  "server",
			Value: `Perl/v?(\d+\.\d+\.\d+)`,
		}},
	},

	OnlySameHost: true,
}

File: src/signatures/technologies/perl.ts
import type { Signature } from "../_types.js";

export const perlSignature: Signature = {
  name: "Perl",
  description:
    "Perl is a family of two high-level, general-purpose, interpreted, dynamic programming languages.",
  cpe: "cpe:2.3:a:perl:perl:*:*:*:*:*:*:*:*",
  rule: {
    confidence: "high",
    headers: {
      Server: "\\bPerl\\b(?: ?/?v?([\\d.]+))?",
    },
  },
};
